### PR TITLE
Fix #215

### LINKED ISSUE #215: fails on imported types from other files
Example code:
```
import type {RowProps} from "./row.jsx";

type StaticProps = RowProps & {
    domain: DomainSlug,
    openLinkInNewTab?: boolean,
};

class TableOfContentsRow extends Component<Props, $FlowFixMeState> {
   ...
}
```

Error:
```
ReferenceError: bpfrpt_proptype_RowProps is not defined
```

I'm going try to create a test case in the test suite that reproduces this.

### PR DESCRIPTION
**Pull request by k4b7**
In some situations where imported prop types were being used in multiple components, some of the AST nodes generated by the plugin were being used in multiple places which caused issues for downstream plugins in `@babel/env`.  This PR fixes that by clone the prop types node before adding it to the AST.

<sub>Resolves #215</sub>

### CODE REVIEW

**Comment by kevinbarabash** [comment]
I ran into some issues when running this on my code so there's at least one edge case I haven't handle.

**Comment by kevinbarabash** [comment]
It turns out I was clone too soon and sometimes `props` would be a function instead of an object.  I've move the clone operation to be right before where the new chunk of AST is inserted into the output tree and that fixed the issue I was seeing.

File: src/__tests__/import-with-private-class-test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+export default class TableOfContentsRowWithQuery extends React.Component<StaticProps> {
+    render() {
+        return <TableOfContentsRow/>;
+    }
+}
+`;
+
+it('import-object', () => {
```
Comment (kevinbarabash) [inline src/__tests__/import-with-private-class-test.js]: I'm not sure how necessary this test case is given there were other tests cases that were exhibiting the issue.  Let me know if you want me to remove this.

File: src/index.js [hunk 2]
```diff
@@ -367,7 +367,9 @@ module.exports = function flowReactPropTypes(babel) {
         return;
       }
 
-      valueNode = propTypesAST;
+      // Clone the props node so that we don't use the same node in multiple
+      // parts of the same AST.
+      valueNode = JSON.parse(JSON.stringify(propTypesAST));
```
Comment (brigand) [inline src/index.js]: I asked in babel slack and there's `babel.types.cloneDeep(propTypesAST)` for this use case.

File: src/__tests__/import-with-private-class-test.js (hunk 1 above)
Comment (brigand) [reply src/__tests__/import-with-private-class-test.js]: The test is good. Let's keep it.

**Review by brigand — CHANGES_REQUESTED** [review]
Just the one cloning issue, and then I'm happy to merge this. Thanks!

**Pull request closed without merging**